### PR TITLE
Fix connection counting

### PR DESCRIPTION
**Pull request by stayallive**
As mentioned in #65, after a reset of the statistics the connection count jumps to the combined amount of users in channels, this is incorrect since they should be counted by their connection (in our case we use the socket id which is unique per connection) because users can be in multiple channels causing the connection count to be very inaccurate in case a lot of channels are used.

### CODE REVIEW

**Comment by stayallive** [comment]
I added a some tests for this (skipped other statistic tests for now to keep the PR small).

While doing this I noticed they connection count is incorrect when the clients are connected but not subscribed to a channel. But that can be fixed later too (this is not new with this PR but the current behaviour).

**Comment by stayallive** [comment]
@mpociot it would be greatly appreciated if this bug can be resolved or I could get some feedback if this PR is something you don't want to accept 👍

**Comment by AlShahawi** [comment]
This pull request helps a lot when limiting the number of connections per application! It needs to be merged ASAP!. Thanks @stayallive

File: tests/TestCase.php [hunk 1]
```diff
@@ -39,12 +45,12 @@ protected function getEnvironmentSetUp($app)
...
-                'id' => 1234,
-                'key' => 'TestKey',
-                'secret' => 'TestSecret',
+                'name'                   => 'Test App',
+                'id'                     => 1234,
+                'key'                    => 'TestKey',
+                'secret'                 => 'TestSecret',
                 'enable_client_messages' => false,
-                'enable_statistics' => true,
+                'enable_statistics'      => true,
```
Comment (francislavoie) [inline tests/TestCase.php]: IMO the whitespace changes in this file aren't necessary, and are inconsistent with the rest of the codebase.

Comment (stayallive) [reply tests/TestCase.php]: You are 100% correct, this is my bad. My editor has a different code style set than this project. Easy fix.

**Comment by mpociot** [comment]
Thank you very much for the PR and your patience :)

**Pull request merged**